src/pkg/bigbang/bigbang.go

```diff
@@ -0,0 +1,368 @@
+package bigbang
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/defenseunicorns/zarf/src/internal/packager/helm"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	helmv2beta1 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1beta2 "github.com/fluxcd/source-controller/api/v1beta2"
+
+	"sigs.k8s.io/yaml"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type bbImageYaml struct {
+	ChartList map[string]bbImageYamlChartDef `json:"package-image-list"`
+}
+
+type bbImageYamlChartDef struct {
+	Version string
+	Images  []string
+}
+
+// Default location for pulling BigBang
+const DEFAULT_BIGBANG_REPO = "https://repo1.dso.mil/big-bang/bigbang.git"
+
+func CreateFluxComponent(bbComponent types.ZarfComponent, bbCount int) (fluxComponent types.ZarfComponent, err error) {
+	fluxComponent.Name = "flux"
+
+	fluxComponent.Required = bbComponent.Required
+
+	fluxManifest := GetFluxManifest(bbComponent.BigBang.Version)
+	fluxComponent.Manifests = []types.ZarfManifest{fluxManifest}
+	repo := DEFAULT_BIGBANG_REPO
+	if bbComponent.BigBang.Repo != "" {
+		repo = bbComponent.BigBang.Repo
+	}
+	images, err := helm.FindFluxImages(repo, bbComponent.BigBang.Version)
+	fluxComponent.Images = images
+
+	return fluxComponent, nil
+}
+
+// MutateBigbangComponent Mutates a component that should deploy BigBang to a set of manifests
+// that contain the flux deployment of BigBang
+func MutateBigbangComponent(componentPath types.ComponentPaths, component types.ZarfComponent) (types.ZarfComponent, error) {
+	_ = utils.CreateDirectory(componentPath.Charts, 0700)
+	_ = utils.CreateDirectory(componentPath.Manifests, 0700)
+
+	repos := make([]string, 0)
+
+	// use the default repo unless overridden
+	if component.BigBang.Repo == "" {
+		repos = append(repos, DEFAULT_BIGBANG_REPO)
+		component.BigBang.Repo = repos[0]
+	} else {
+		repos = append(repos, fmt.Sprintf("%s@%s", component.BigBang.Repo, component.BigBang.Version))
+	}
+
+	// download bigbang so we can peek inside
+	chart := types.ZarfChart{
+		Name:        "bigbang",
+		Namespace:   "bigbang",
+		URL:         repos[0],
+		Version:     component.BigBang.Version,
+		ValuesFiles: component.BigBang.ValuesFrom,
+		GitPath:     "./chart",
+	}
+
+	zarfHelmInstance := helm.Helm{
```

Comment:
We usually call this helmCfg elsewhere and would be nice to maintain consistency.